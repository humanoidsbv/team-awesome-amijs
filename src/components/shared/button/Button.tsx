import React from "react";

import * as Styled from "./Button.styled";

interface ButtonProps {
  isFormVisible: boolean;
  label?: string;
  onClick: Function;
  type?: "Primary" | "Secondary" | "Tertiary";
  icon: SVGElement;
}

function Button({ isFormVisible, label, onClick, type, icon }: ButtonProps) {
  return type === "Primary" ? (
    <Styled.Primary {...{ isFormVisible, onClick, type }}>
      {icon}
      {label}
    </Styled.Primary>
  ) : (
    <Styled.Secondary {...{ isFormVisible, onClick, type, icon }}>
      {icon}
      {label}
    </Styled.Secondary>
  );
}

export default Button;
