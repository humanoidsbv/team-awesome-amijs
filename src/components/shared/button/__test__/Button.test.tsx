import React from "react";
import renderer from "react-test-renderer";
import Button from "../Button";

it("renders a button", () => {
  const tree = renderer
    .create(<Button onClick={() => {}} isFormVisible type="Primary" label="test" icon />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
