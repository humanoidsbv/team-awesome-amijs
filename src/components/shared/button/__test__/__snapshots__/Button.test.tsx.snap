// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a button 1`] = `
<button
  className="Buttonstyled__Primary-sc-1fwfac-0 bLoHSc"
  onClick={[Function]}
  type="Primary"
>
  test
</button>
`;
